src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1261,6 +1274,28 @@ MultiShardModifyQuerySupported(Query *originalQuery,
 }
 
 
+/*
+ * SingleShardModifyQuerySupported returns the error message if the modify query is
+ * not routable, otherwise it returns NULL.
+ */
+static DeferredErrorMessage *
+SingleShardModifyQuerySupported(Query *originalQuery,
+								PlannerRestrictionContext *plannerRestrictionContext)
+{
+	DeferredErrorMessage *errorMessage = NULL;
+	if (FindNodeMatchingCheckFunction((Node *) originalQuery,
```

Comment:
Maybe add a comment on why we're disallowing volatile functions:

We currently do not support volatile functions in update/delete statements because the function evaluation logic does not know how to distinguish volatile functions (that need to be evaluated per row) from stable functions (that need to be evaluated per query), and it is also not safe to push the volatile functions down on replicated tables.